File: lockena-miniapp/src/api/dto/vault-item/vault-item.dto.ts
export interface VaultItemDto {
  id: string;
  encryptedItemKey: string;
  itemKeyIv: string;
  encryptedPayload: string;
  payloadIv: string;
  blindIndex: string;
  createdAtUtc: string;
  updatedAtUtc: string;
}

File: Lockena.Backend/Lockena.Application/DTO/VaultItem/VaultItemDto.cs
namespace Lockena.Application.DTO.VaultItem
{
    public class VaultItemDto
    {
        public Guid Id { get; set; }
        public string EncryptedItemKey { get; set; }
        public string EncryptedPayload { get; set; }
        public DateTime CreatedAtUtc { get; set; }
        public DateTime UpdatedAtUtc { get; set; }

        public VaultItemDto(Domain.Entities.VaultItem item)
        {
            Id = item.Id;
            EncryptedItemKey = item.EncryptedItemKey;
            EncryptedPayload = item.EncryptedPayload;
            CreatedAtUtc = item.CreatedAtUtc;
            UpdatedAtUtc = item.UpdatedAtUtc;
        }
    }
}
